src/GitVersion.Core/VersionCalculation/VersionSearchStrategies/TaggedCommitVersionStrategy.cs

```diff
@@ -50,6 +53,19 @@ private IEnumerable<BaseVersion> GetBaseVersionsInternal(EffectiveBranchConfigur
                 continue;
             }
 
+            var alternativeSemanticVersionMax = alternativeSemanticVersionsWithTag.Max()?.Value;
+            var highestPossibleSemanticVersion = semanticVersionWithTag.Value.Increment(
+                VersionField.Major, null, forceIncrement: true, alternativeSemanticVersionMax
+            );
+            if (highestPossibleSemanticVersion.IsLessThan(semanticVersionTreshold, includePreRelease: false))
+            {
+                this.log.Info(
+                    $"The tag '{semanticVersionWithTag.Value}' is skipped because it will never be higher than other tags."
```

Comment:
I think "never be higher" needs elaboration. I've made an attempt, but please correct me if I've misunderstood.

```suggestion
                    $"The tag '{semanticVersionWithTag.Value}' is skipped because it provides a lower base version than other tags."
```